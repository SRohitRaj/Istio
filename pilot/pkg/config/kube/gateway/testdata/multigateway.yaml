apiVersion: networking.x-k8s.io/v1alpha1
kind: GatewayClass
metadata:
  name: istio
spec:
  controller: istio.io/gateway-controller
  parametersRef:
    group: core
    kind: ConfigMap
    name: istio-gateway-class
    scope: Namespace
    namespace: istio-system
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: GatewayClass
metadata:
  name: istio-cluster-local
spec:
  controller: istio.io/gateway-controller
  parametersRef:
    group: core
    kind: ConfigMap
    name: istio-cluster-local-gateway-class
    scope: Namespace
    namespace: istio-system
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: Gateway
metadata:
  name: ingress-gateway
  namespace: istio-system
spec:
  gatewayClassName: istio
  listeners:
  - port: 34000
    protocol: TCP
    routes:
      namespaces:
        from: All
      kind: TCPRoute
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: Gateway
metadata:
  name: cluster-local-gateway
  namespace: istio-system
spec:
  gatewayClassName: istio-cluster-local
  listeners:
  - port: 34000
    protocol: TCP
    routes:
      namespaces:
        from: All
      kind: TCPRoute
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: TCPRoute
metadata:
  name: tcp
  namespace: default
  labels:
    selected: "yes"
spec:
  gateways:
    allow: FromList
    gatewayRefs:
    - name: cluster-local-gateway
      namespace: istio-system
    - name: ingress-gateway
      namespace: istio-system
  rules:
  - forwardTo:
    - serviceName: httpbin
      port: 9090
