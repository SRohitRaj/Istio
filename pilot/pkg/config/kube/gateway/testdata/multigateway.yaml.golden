apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  creationTimestamp: null
  name: ingress-gateway-istio-autogenerated-k8s-gateway
  namespace: istio-system
spec:
  selector:
    istio: ingressgateway
  servers:
  - hosts:
    - '*'
    port:
      name: tcp-34000-gateway-ingress-gateway-istio-system
      number: 34000
      protocol: TCP
---
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  creationTimestamp: null
  name: cluster-local-gateway-istio-autogenerated-k8s-gateway
  namespace: istio-system
spec:
  selector:
    istio: cluster-local
  servers:
  - hosts:
    - '*'
    port:
      name: tcp-34000-gateway-cluster-local-gateway-istio-system
      number: 34000
      protocol: TCP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: tcp-tcp-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/ingress-gateway-istio-autogenerated-k8s-gateway
  - istio-system/cluster-local-gateway-istio-autogenerated-k8s-gateway
  hosts:
  - '*'
  tcp:
  - route:
    - destination:
        host: httpbin.default.svc.cluster.local
        port:
          number: 9090
---
